src/math/web_mercator.rs

```diff
@@ -0,0 +1,146 @@
+//! Calculations with Web Mercator coordinates and tiles.
+
+use nalgebra::Vector2;
+use nav_types::WGS84;
+use std::f64::consts::{FRAC_1_PI, PI};
+
+/// 2.0 * E.powf(PI).arctan() - FRAC_PI_2;
+/// In degrees, it's 85.051129 (cf. Wikipedia)
+const LAT_BOUND_RAD: f64 = 1.48442222974533244439498957945034;
+
+/// LAT_BOUND_SIN = sin(LAT_BOUND_RAD)
+const LAT_BOUND_SIN: f64 = 0.99627207622074998027983383508399;
+
+const TWO_PI: f64 = 2.0 * PI;
+const FOUR_PI: f64 = 4.0 * PI;
+const FOUR_PI_INV: f64 = 0.25 * FRAC_1_PI;
+const TILE_SIZE: u32 = 256;
+
+/// The max zoom level is currently 23 because it makes implementation easier,
+/// but theoretically nothing stops us from going deeper.
```

Comment:
the current max zoom is given by the TILE_SIZE<<zoom which is expressed in u32 in line 81. I'd say it is currently limited by the implementation choice, or make this constant bound to the integer bits that describe TILE_SIZE.

Reply:
I edited the comment.